app/controllers/spina/registers_controller.rb

```diff
@@ -73,11 +73,52 @@ def history
         { current_record: current_record, previous_record: previous_record, updated_fields: changed_fields, key: entry[:key] }
       }
 
-      @entries_with_items = paginate(entries_mapped_with_items)
+      if params[:search_input].present?
+        filtered = filter(entries_mapped_with_items, params[:search_input])
+        @entries_with_items = paginate(filtered)
+      else
+        @entries_with_items = paginate(entries_mapped_with_items)
+      end
     end
 
     private
 
+    def filter(entries_mapped_with_items, filter_value)
+      entries_mapped_with_items.select { |entry|
+        entry[:key].downcase.include?(filter_value.downcase) ||
+          contained?(entry[:previous_record], filter_value) ||
+          contained?(entry[:current_record], filter_value)
+      }
+    end
+
+    def contained?(item, request_value)
```

Comment:
Please can we merge this method with the existing `search` method? They are attempting to do the same thing and it would be good to de-duplicate this logic before merging.

I realise that the existing `search` method fails for an item with cardinality `n` field, so this would also fix the cardinality `n` problem for searching on the new records page.